test/src/test/java/hudson/model/utils/AbortExceptionPublisher.java

```diff
@@ -0,0 +1,35 @@
+package hudson.model.utils;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Publisher;
+import hudson.tasks.Recorder;
+
+import java.io.IOException;
+
+/**
+ * Publisher that throws AbortException
+ */
+public class AbortExceptionPublisher extends Recorder {
```

Comment:
I notice these are all `Recorder`s, none `Notifier`s which run in a later build phase and which are not expected to affect the `result`. Not sure if it matters for purposes of this PR, just something to consider.